src/backend/app/tools.py

```diff
@@ -61,19 +62,79 @@ def generate_prediction(eval_params):
         {h3.geo_to_h3(lat, lon, hex_resolution) for lat, lon in coordinates}
     )
 
-    hull = alphashape.alphashape(coordinates, 1)
-    hull_points = list(mapping(hull)["coordinates"])
+    # hull = alphashape.alphashape(coordinates, 1)
+    # hull_points = list(mapping(hull)["coordinates"])
 
     annotation_hexagon_ids = {
         "presence": prediction_hexagon_ids,
         "absence": list()
     }
 
     return dict(
-        coordinates=coordinates.tolist(),
-        pred_loc_combined=pred_loc_combined.tolist(),
-        hull_points=hull_points,
+        # coordinates=coordinates.tolist(),
+        # pred_loc_combined=pred_loc_combined.tolist(),
+        # hull_points=hull_points,
         prediction_hexagon_ids=prediction_hexagon_ids,
         # set prediction_hexagon_ids as the starting point for the annotation
         annotation_hexagon_ids=annotation_hexagon_ids,
     )
+
+
+def generate_prediction_scores(eval_params, lowest_threshold=0.0001):
+    preds, locs = get_prediction(eval_params)
+
+    # switch lats and longs
+    locs[:, [1, 0]] = locs[:, [0, 1]]
+
+    # combine coordinates and predictions
+    pred_loc_combined = np.column_stack((locs, preds))
+    pred_loc_combined = np.float_(pred_loc_combined)
+    pred_loc_combined = pred_loc_combined[pred_loc_combined[:, 2] >= lowest_threshold]
+    hex_resolution = eval_params['hex_resolution']
+
+    # if a more detailed HeatMap needed, use `pred_loc_combined` for that
+    scores_dict={}
+    for lat, lon, pred in pred_loc_combined:
+        try:
+            scores_dict[h3.geo_to_h3(lat, lon, hex_resolution)].append(pred)
+        except Exception:
+            scores_dict[h3.geo_to_h3(lat, lon, hex_resolution)]=[pred]
+    hex_indexes=[]
+    hex_scores=[]
+    for key in scores_dict:
+        hex_indexes.append(key)
+        hex_scores.append(max(scores_dict[key]))
+    
+    index_score_combined = np.column_stack((hex_indexes, hex_scores))
```

Comment:
and here, if we already have only max scores in our ` scores_dict`, then we can do it like this
```python
return np.column_stack((scores_dict.keys(), scores_dict.values()))
```

but i think we can use `scores_dict` in dict format later. No need to convert it to np table.

Reply:
In this case, scores_dict has all the values in the form:
'872fffffff' : [0.123, 0.543, 0.345]
so scores_dict is storing all values, not the maximum one, therefore 
return np.column_stack((scores_dict.keys(), scores_dict.values()))
would not return the max to each hexagon index. 

I realize that this part of the code works fast and properly, but it's not really elegant. I'll try to simplify it.